src/engine/source/kvdb/src/kvdbHandler.cpp

```diff
@@ -144,20 +144,51 @@ std::variant<std::string, base::Error> KVDBHandler::get(const std::string& key)
     return base::Error {"Cannot access RocksDB::DB"};
 }
 
-std::variant<std::unordered_map<std::string, std::string>, base::Error> KVDBHandler::dump()
+std::variant<std::list<std::pair<std::string, std::string>>, base::Error> KVDBHandler::dump(const uint32_t page,
+                                                                                            const uint32_t records)
 {
     auto pRocksDB = m_weakDB.lock();
     if (pRocksDB)
     {
         auto pCFhandle = m_weakCFHandle.lock();
         if (pCFhandle)
         {
-            std::unordered_map<std::string, std::string> content {};
+            std::list<std::pair<std::string, std::string>> content;
             std::shared_ptr<rocksdb::Iterator> iter(pRocksDB->NewIterator(rocksdb::ReadOptions(), pCFhandle.get()));
+            uint32_t actualPage = 1, counterRecords = 1;
+
+            // std::string aproxSizeProp {};
+            // pRocksDB->GetProperty("rocksdb.estimate-num-keys", &aproxSizeProp);
+            // std::cout<<"Total size: "<<aproxSizeProp<<std::endl;
 
-            for (iter->SeekToFirst(); iter->Valid(); iter->Next())
+            if (page == 0 && records == 0)
             {
-                content[iter->key().ToString()] = iter->value().ToString();
+                for (iter->SeekToFirst(); iter->Valid(); iter->Next())
+                {
+                    content.push_back(std::make_pair(iter->key().ToString(), iter->value().ToString()));
+                }
+            }
+            else
+            {
+                for (iter->SeekToFirst(); iter->Valid() && actualPage <= page; iter->Next())
+                {
+                    if (actualPage == page)
+                    {
+                        std::cout << "Iter: " << iter->key().ToString() << " : " << iter->value().ToString()
```

Comment:
why we output to cout here?

Reply:
This should be removed, it was for testing purposes.